internal/helm/helmdriver/transport_http.go

```diff
@@ -0,0 +1,153 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helmdriver
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"emperror.dev/errors"
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/gorilla/mux"
+	kitxhttp "github.com/sagikazarmark/kitx/transport/http"
+
+	apphttp "github.com/banzaicloud/pipeline/internal/platform/appkit/transport/http"
+)
+
+func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, options ...kithttp.ServerOption) {
+	errorEncoder := kitxhttp.NewJSONProblemErrorResponseEncoder(apphttp.NewDefaultProblemConverter())
+
+	router.Methods(http.MethodPost).Path("").Handler(kithttp.NewServer(
+		endpoints.AddRepository,
+		decodeAddRepositoryHTTPRequest,
+		kitxhttp.ErrorResponseEncoder(encodeCreateRepositoryHTTPResponse, errorEncoder),
+		options...,
+	))
+
+	router.Methods(http.MethodGet).Path("").Handler(kithttp.NewServer(
+		endpoints.ListRepositories,
+		decodeListRepositoriesHTTPRequest,
+		kitxhttp.ErrorResponseEncoder(encodeListRepositoriesHTTPResponse, errorEncoder),
+		options...,
+	))
+
+	router.Methods(http.MethodDelete).Path("/{repoName}").Handler(kithttp.NewServer(
+		endpoints.DeleteRepository,
+		decodeDeleteRepositoryHTTPRequest,
+		kitxhttp.ErrorResponseEncoder(encodeDeleteRepositoryHTTPResponse, errorEncoder),
+		options...,
+	))
+}
+
+func decodeAddRepositoryHTTPRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	orgID, e := extractOrgID(r)
+	if e != nil {
+		return nil, errors.WrapIff(e, "failed to decode add repository request")
+	}
+
+	var addRepositoryRequest AddRepositoryRequest
+
+	err := json.NewDecoder(r.Body).Decode(&addRepositoryRequest)
```

Comment:
By looking at the code the `AddRepositoryRequest` looks like this:

```go
// AddRepositoryRequest is a request struct for AddRepository endpoint.
type AddRepositoryRequest struct {
	OrganizationID uint
	Repository     helm.Repository
}
```

which would require the following JSON payload (which is probably not the right one):

```json
{
    "Repository": {
        "Name": "..."
    },
}
```

(correct me if I'm wrong).

Generally you should unmarshal into a struct generated by openapi, and manually convert the data to the internal type (at least I don't see how the generated types could be used here)

Same applies to the decoders below.

Reply:
thx fixed